src/util/support/threads.c

```diff
@@ -237,7 +237,7 @@ void *k5_getspecific (k5_key_t keynum)
     if (err)
         return NULL;
 
-    assert(keynum >= 0 && keynum < K5_KEY_MAX);
+    assert(keynum < K5_KEY_MAX);
```

Comment:
clang presumably chooses an unsigned type for the enumeration, and doesn't remember enough state to relax the warning on the grounds that the type might be signed.  We can just get rid of the assert; it's not likely that code would pass an invalid key value.